autogtp/main.cpp

```diff
@@ -69,6 +69,9 @@ int main(int argc, char *argv[]) {
     QCommandLineOption singleOption(
         { "s", "single" }, "Exit after the first game is completed.",
                           "");
+    QCommandLineOption watcherOption(
+        { "w", "watcher" }, "Reject boring 0 resignation games.",
```

Comment:
I really don't like this. 0 resignation games are essential for the learning to detect cases where it's highly convinced of the evaluation but it's wrong, and there's anecdotal evidence from other projects (aside from the published AGZ runs) that having a fair amount of them is required for good performance.

Reply:
The server hands them out statistically, I think. So clients purposefully ignoring games will eventually cause things to go whack.

>Either way, my experience using LeelaWatcher greatly improved after this change, and I will not go back.

🙄

Reply:
The decision is simply made at the time a request is made. It isn't tracked to be sure we're getting a property # returned. So if a lot of people were using LeelaWatcher, we'd have less no-resign games arriving in the database. The missing ones wouldn't be re-requested to other clients.

Reply:
Then changing the server code to account for this is needed. I will take a look and see if a fix is easy enough for me to care about doing.

Reply:
I found the part where resignation_percent is assigned, yet the database structure seems to need modification to include network1_0r_wins, network1_0r_losses. SPRT(match.network1_0r_wins, match.network1_0r_losses) may also be a useful indicator.

Reply:
Quoting 'Mastering the Game of Go without Human Knowledge': 'The resignation threshold v_resign is selected automatically to keep the fraction of false positives (games that could have been won if AlphaGo had not resigned) below 5%. To measure false positives, we disable resignation in 10% of self-play games and play until termination.'

server.js implements 20%(probably) rather than 10%(guaranteed):
if (Math.random() < 0.2) options.resignation_percent = "0";

Things will be fine(probably).

Reply:
I'd prefer if this was taken out. Would surely make this a lot easier to pull.